odoo/addons/base/models/res_currency.py

```diff
@@ -171,11 +170,12 @@ def is_zero(self, amount):
         self.ensure_one()
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
-    @api.model
+    @tools.ormcache('from_currency', 'to_currency', 'company', 'date')
```

Comment:
@antonylesuisse you're aware that this creates *more* causes for cache invalidation, when you're in the process of chasing them down, right? The dashboard speed is not impacted by this in our case, as there are only 3 calls in total, and we cannot measure any positive speed difference.

However it is quite possible that it helps in other cases where we have more repeated calls to `_get_conversion_rate()` within a single transaction. It would be interesting to find them though, as that would help defend this against a future removal ;-)

Reply:
Yes maybe we dont need that one as it's not really helpful for odoo.com, otherwise the cron frequency for currency update (currency_interval_unit) is manually by default and daily at most.

Reply:
The vast majority of currency rate computation is done in loops. It is not always easy to implement a local cache. (often used in functions with `self.ensure_one()` called inside of loops)

Reply:
@odony @antonylesuisse What do we do with this?
Note that this will cause cache invalidation at most once per day, in our case it is once per month.
However, it will improve the performance of a lot of parts in the code since the rates are often computed multiple times.

Should I do some benchmark?

Reply:
@antonylesuisse Monthly reminder to ask you what is your final opinion on this

Reply:
@antonylesuisse Monthly reminder to ask you what is your final opinion on this

Reply:
@antonylesuisse Monthly reminder to ask you what is your final opinion on this

Did you blacklist me already?